src/bindings/lib/bindings.c

```diff
@@ -52,21 +52,33 @@ static void finalize_tree(value v) {
 }
 
 static struct custom_operations tree_custom_ops = {
-  identifier : "tree handling",
-  finalize : finalize_tree,
-  compare : custom_compare_default,
-  hash : custom_hash_default,
-  serialize : custom_serialize_default,
-  deserialize : custom_deserialize_default
+  .identifier = "tree handling",
+  .finalize = finalize_tree,
+  .compare = custom_compare_default,
+  .hash = custom_hash_default,
+  .serialize = custom_serialize_default,
+  .deserialize = custom_deserialize_default,
+#ifdef custom_compare_ext_default
+  .compare_ext = custom_compare_ext_default,
+#endif
+#ifdef custom_fixed_length_default
+  .fixed_length = custom_fixed_length_default,
+#endif
```

Comment:
Where do the new fields under ifdefs come from? I expect this code to be copied from the original tree-sitter bindings for Reason. The project I found is oni2 and doesn't have the ifdefs: https://github.com/onivim/oni2/blob/161a92c2226728b37cd9a6f554b7397f4beb3d5e/src/reason-tree-sitter/bindings.c#L45-L61

Can you please include an authoritative source URL at the beginning of this file?

Reply:
There's also https://github.com/onivim/reason-tree-sitter/blob/master/src/bindings.c but the project was archived.

Reply:
They were introduced in OCaml 5.0, see [`caml/custom.h`](https://github.com/ocaml/ocaml/blob/fb3bac9f88a4eb7a48f347ca25005cc4e51828ea/runtime/caml/custom.h#L27-L46). Using the `#ifdef` allows the code to remain compatible with OCaml 4 (one could also use the `OCAML_VERSION` macro, but I find this pattern rather elegant). Some compilers in some settings complain if some fields are left uninitialized.